vignettes/generate-color-palette.Rmd

```diff
@@ -0,0 +1,80 @@
+---
+title: "palettizer: Color palette generation from images"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{generate-color-palette}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>",
+  fig.path = "man/figures/",
+  out.width = "100%"
+)
+```
+
+This vignette demonstrates how to use the {palettizer} package to generate a color palette from an image and apply it to a {ggplot2} plot using the palmerpenguins dataset.
+
+First, load the required libraries:
+
+```{r load-libraries}
+library(palettizer)
+library(palmerpenguins)
+library(ggplot2)
+```
+
+## Generate color palette
+
+The {palettizer} package generates a color palette from an image using K-means clustering. The algorithm works as follows:
+
+1. The image is first resized to a smaller version for faster processing.
+2. The colors of the image are then represented in the LAB color space, which is more perceptually uniform compared to RGB.
+3. K-means clustering is performed on the colors in the LAB color space. The number of clusters (K) corresponds to the desired number of colors in the palette, which can be specified using the `cluster_count` parameter.
+4. The centroids of the K clusters are calculated, representing the dominant colors in the image.
+5. The dominant colors are converted back to the RGB color space and returned as a vector of hexadecimal color codes.
+
+To generate a color palette from an image using the `plt_tize()` function, provide the image file path and the desired number of colors (clusters):
+
+```{r generate-palette}
+plt_vector <-
+  palettizer::plt_tize("lter_penguins.png", cluster_count = 3)
```

Comment:
I don't think you need the palettizer:: since you've already loaded the package, and this can all likely go in a single line.